aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -119,22 +119,17 @@ std::vector<at::Tensor> NestedTensor_unbind(
       "got dimension ",
       dim,
       " instead.");
-  auto esizes = get_nested_size_tensor(self);
-  std::vector<at::Tensor> result_tensors;
-  if (esizes.dim() == 0) {
-    return result_tensors;
-  }
-  auto esizes_chunks = esizes.unbind(0);
-  std::vector<int64_t> splits;
-  for (const auto i : c10::irange(esizes_chunks.size())) {
-    splits.push_back(esizes_chunks[i].prod().item<int64_t>());
+  auto self_ptr = get_nested_tensor_impl(self);
```

Comment:
Hey @cpuhrsch, as you mentioned before in [#78934](https://github.com/pytorch/pytorch/pull/78934), the unbind function can avoid some overhead by operating on the nested size directly. Since we added the batch dimension offset utility (see last comment), I refactored unbind to make use of it